yoti-sdk-impl/src/main/java/com/yoti/api/client/shareurl/constraint/SimpleSourceConstraintBuilder.java

```diff
@@ -0,0 +1,39 @@
+package com.yoti.api.client.shareurl.constraint;
+
+import com.yoti.api.client.shareurl.policy.WantedAnchor;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class SimpleSourceConstraintBuilder implements SourceConstraintBuilder {
+
+    private List<WantedAnchor> wantedAnchors;
+
+    private boolean softPreference;
+
+    SimpleSourceConstraintBuilder() {
+        this.wantedAnchors = new ArrayList<>();
+        this.softPreference = false;
+    }
+
+    @Override
+    public SourceConstraintBuilder withWantedAnchor(WantedAnchor wantedAnchor) {
+        this.wantedAnchors.add(wantedAnchor);
+        return this;
+    }
+
+    @Override
+    public SourceConstraintBuilder withSoftPreference(boolean softPreference) {
+        this.softPreference = softPreference;
+        return this;
+    }
+
+    @Override
+    public SourceConstraint build() {
+        return new SimpleSourceConstraint(
+                Collections.unmodifiableList(wantedAnchors),
```

Comment:
Not sure if we need to enforce this being unmodifiable here - this is just for the client building the request to be sent to us, I don't think we mind if it's altered before it's sent to us (unlike the profile objects that we return), would you agree?

Reply:
As Lists are passed by reference, would prefer to keep it unmodifiable once the WantedAttribute has been built 👍